### ISSUE TITLE
Provide simple one-time setting of environment variable

### ISSUE THREAD

**Issue by lzybkr** [PowerShell/PowerShell#3316]
Instructions for many tools from the Linux world suggest setting an environment variable for a single invocation of a command like:

```bash
JEKYLL_ENV=production jekyll build
```

The equivalent in PowerShell is cumbersome:

```PowerShell
try {
    $oldValue = $env:JEKYLL_ENV
    $env:JEKYLL_ENV = "production"
    jekyll build
} finally {
    $env:JEKYLL_ENV = $oldValue
}
```

I believe a similar syntax could work in PowerShell:

```PowerShell
$env:JEKYLL_ENV="production" jekyll build
```

**Comment by andschwa** [comment]
This would be a great usability enhancement. A lot of our tests rely on the current cumbersome method because of the lack of this feature.

**Comment by FranklinYu** [comment]
It's easier to implement this as an argument to `Start-Process` like

```powershell
Start-Process main.exe -Environment @{ FOO = 'bar' }
```

The source code is in

https://github.com/PowerShell/PowerShell/blob/0cc1f06be4c2f5f46852fec65549eb9260555298/src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs#L1574-L1579

**Comment by johnjelinek** [comment]
>Start-Process : A parameter cannot be found that matches parameter name 'Environment'.

**Comment by lzybkr** [comment]
Just yesterday I came up with a [proof of concept](https://gist.github.com/lzybkr/80c66be2bd866a2e7387b83045e0ab81) by abusing the CommandNotFoundHandler.

**Comment by mklement0** [comment]
@johnjelinek: Adding an `-Environment` parameter to `Start-Process` is a _proposal_ at this point.

@FranklinYu:

Even though `Start-Process` too should support passing arbitrary environment variables, as proposed in #4761, it is not a substitute for @lzybkr's proposal:

Aside from being more verbose, `Start-Process`-launched processes aren't connected to PowerShell's output streams, even when you use `-Wait` and `-NoNewWindow`, so you cannot directly capture or redirect / suppress output.

@lzybkr: 

Love the idea; just to spell it out: it should work for  _multiple_ env. variables too, as in POSIX-like shells:

```sh
# In a POSIX-like shell such as Bash
$ foo=1 bar=2 sh -c 'echo $foo $bar'  # -> '1 2'
```

Presumed PS equivalent:

```powershell
PS> $env:foo=1 $env:bar=2 sh -c 'echo $foo $bar'  # -> '1 2'
```

Given that this a feature for _child processes_ (external programs), in the context of which only _environment_ variables can be meaningfully defined, perhaps even omitting the `env:` part is an option - though perhaps that is conceptually too confusing:

```powershell
# Implied `env:`?
PS> $foo=1 $bar=2 sh -c 'echo $foo $bar'  # -> '1 2'
```

**Comment by vexx32** [comment]
Hmm. That would be nice, but I feel like the syntax is a bit obtuse. Maybe something like this?
```powershell
with @{Path = $NewPath; JEKYLL_ENV = $jekyll} & $command $params 
```
That would allow use of multiline syntax quite naturally for less lengthy lines and better readability:

```powershell
with @{
    Path = $NewPath
    JEKYLL_ENV = $jekyll
} & $command $params 
```
It also has a direct tie-in to the existing `&` operator for a bit better indication of what's actually happening.

**Comment by lzybkr** [comment]
The hashtable syntax isn't horrible - that's what I ended up with [here](https://gist.github.com/lzybkr/fa46d9d50b801ae9cb66ad2bb65f0ef6) - invoked like:

```powershell
Invoke-WithEnvironment @{ JEKYLL_ENV="production" } { jekyll build }
```

**Comment by SteveL-MSFT** [comment]
Would it make sense to simply add a `-Environment @{}` parameter to `Invoke-Command`?

**Comment by vexx32** [comment]
Adding `-Environment` to both Start-Process and Invoke-Command are both good options, I think. 🙂

**Comment by iSazonov** [comment]
I suggest to discuss the parameters in #4671.

**Comment by mklement0** [comment]
_Also_ making `Invoke-Command` support `-Environment @{}`, alongside `Start-Process`, as proposed in #4671, sounds like a good idea.

However, at least to me, this proposal is about a syntax that is:

* (a) low on ceremony (terse)

* (b) doesn't require a cmdlet call, i.e. is a _language feature_

`Invoke-Command @{ ... } { ... }` fits neither bill.

@vexx32's proposal definitely meets (b); with respect to (a): while  definitely more PowerShell-like, it still feels a bit verbose, especially for the - presumably common - case of passing a _single_ env.-var value.

If, by contrast, we go all in, we could _simply mimic POSIX syntax as-is_:

```
PS> foo=1 bar=2 sh -c 'echo $foo $bar'   # -> '1 2'
```

While a tad obscure, you could argue that we're entering a different world here anyway - that of _external programs_ - where different rules already apply (such as `--%`, how arrays are passed, ...).

**Comment by SteveL-MSFT** [comment]
The obscure syntax creates a discovery problem.  Interactive, you can use a more terse syntax:

```powershell
icm -env @{ foo=1; bar=2 } sh -c 'echo $foo $bar'
```

**Comment by rkeithhill** [comment]
Given there is also an issue with PS effectively ignoring a native app's exit code, maybe this could be combined into a command like `Start-NativeExecution` where one of the parameters would be `-Environment` and, like in the PS build module, it throws on a non-zero exit code.  

When invoking native apps you often what this "throw on non-zero exit code" functionality.  I can't tell you how many times I've helped folks figure out why their build was passing when the log shows obvious errors.

**Comment by vexx32** [comment]
Eh, I mean... that's kinda just duplicating what Start-Process could do with an extra parameter or two. I'm not sure how much sense it makes to have _two_ commands in that space at the moment? 🤔

**Comment by rkeithhill** [comment]
`Start-Process` doesn't help with the test for $LASTEXITCODE and often in these CI/build/test scenarios you want a failed native app invocation to fail the build/test.

**Comment by vexx32** [comment]
You could easily enhance Start-Process to accept a parameter (switch? int[]?) to throw on any non-zero or a predefined list of "bad" exit codes, though; that doesn't really mandate an entirely new command, I would think.

**Comment by mklement0** [comment]
Note that `&&` and `||` were specifically introduced for these scenarios, enabling you to write, for instance:

```powershell
foo -bar || exit 1  
# or 
foo -bar || throw
```

Unfortunately, that currently doesn't actually work, because you're required to wrap the `exit` / `throw` / `return` in `$(...)`, which is  both unexpected and cumbersome.

```powershell
foo -bar || $(throw)
```

The ship hasn't sailed yet, fortunately, so I invite you to join the conversation at #10967.

**Comment by mklement0** [comment]
@SteveL-MSFT:

> Interactive, you can use a more terse syntax:

I think it would be beneficial to provide something is _robustly_ terse, and doesn't rely on aliases.

> The obscure syntax creates a discovery problem

That could be remedied with a dedicated conceptual help topic covering invocation of native apps, which is definitely urgently needed even with the current capabilities - see https://github.com/MicrosoftDocs/PowerShell-Docs/issues/5152

**Comment by SteveL-MSFT** [comment]
I don't think the addition to Invoke-Command (or Start-Process) is mutually exclusive to something terse.  If we wanted to explore that route, my preference is to use Bash syntax exactly so user would be able to cut and paste most examples (barring other issues we have with arg parsing/passing...) and have them just work in PowerShell.

**Comment by mklement0** [comment]
> my preference is to use Bash syntax exactly so user would be able to cut and paste most examples

I agree that this would be best; to recap what that looks like:

```powershell
# Define command-scoped (child-process-only) $env:foo and $env:bar env. variables
foo=1 bar=2 sh -c 'echo $foo $bar'   # -> '1 2'
```
> I don't think the addition to Invoke-Command (or Start-Process) is mutually exclusive to something terse

Agreed, and #4671 already proposes that for `Start-Process` (introducing an `-Environment` parameter accepting a hashtable).

With `Invoke-Command` we have to be careful, though, because there is no guarantee that a _child process_ is involved in whose environment the environment-variable definitions should - ephemerally - take effect.

I think it makes sense to always restrict the specified environment variables to _child processes_, which means that the _current_ process' environment is _not_ modified - that's how it (sensibly) works in POSIX-like shells.

Therefore, in the following invocation the Powershell code inside the `-ScriptBlock` parameter would _not_ see `$env:foo`, which invites confusion.

```powershell
# $env:foo is NOT defined in the script block.
Invoke-Command -Environment @{ foo = 'bar' } -ScriptBlock { "`$env:foo is: $env:foo" }
```

In short: I think it's sufficient to offer the Bash-like syntax specifically for _external programs_ and otherwise enhance `Start-Process` only.

**Comment by TylerLeonhardt** [comment]
A real life example where they use this concept in bash:

https://www.twilio.com/blog/2017/08/working-with-environment-variables-in-node-js.html

```
PORT=9999 node server.js
```

**Comment by iSazonov** [comment]
First step was done in #10830.
Now we need to add new parameter - `-Environment`.
Then we could resolve the issue.
But we have over 90 opened PRs so I stopped the work.

**Comment by yinso** [comment]
If the proposal is a `-Environment` parameter - want to drop a line noting that the unix approach makes a distinction between environment variables and command line parameters, in that the env variables aren't tracked by the process manager and hence aren't visible when someone inspects the process.

This is often used to pass more sensitive data such as passwords. When someone else runs a `ps aux` you don't have to worry about the password being leaked.

I don't know whether the process manager in Windows has the same issue, but it would be great to consider for this use case, thanks.

**Comment by iSazonov** [comment]
@yinso Can you point guides/docs about Unix behavior/implementation?

**Comment by yinso** [comment]
@iSazonov sorry I missed your message. I am not certain whether there is an official documentation for this behavior, but below is what I found:

* According to wikipedia, environment variable was introduced [by Version 7 Unix in 1979, and all subsequent variants, as well as DOS since 1982](https://en.wikipedia.org/wiki/Environment_variable) has this capability.
* As we all know, environment variables are passed into processes as part of its invocation. Captured in the same wikipedia page is [a quote that says](https://en.wikipedia.org/wiki/Environment_variable#Design):
  > Alternatively, from command shells such as bash, a user can change environment variables for a particular command invocation by indirectly invoking it via `env` or using the `ENVIRONMENT_VARIABLE=VALUE <command>` notation.
* from there we can infer that this is a capability that belongs to the individual shells. Looking up [bash's documentation, we found the following](https://www.gnu.org/software/bash/manual/bash.html#Environment):
  > The environment for any simple command or function may be augmented temporarily by prefixing it with parameter assignments, as described in Shell Parameters. These assignment statements affect only the environment seen by that command.

Based on the above, my interpretation is that the `ENV_VAR1=<val1> ENV_VAR2=<val2> <cmd ...>` is a syntax that bash can specifically parse, so that the current environment variables are augmented with `ENV_VAR1` and `ENV_VAR2` to be passed to `cmd ...`, so that once parsed, the `ENV_VAR1` and `ENV_VAR2` aren't part of the command, and thus not captured in the process table.

Hope this helps, let me know if there are any questions thanks.

**Comment by edgetools** [comment]
curious how this proposal intersects with the bash-style invocation examples provided above?

currently we do not have to use `&` or `Invoke-Command` or `Start-Process` to invoke a process, similar to how we would execute it on the linux command line

this is a completely accepted way of invoking commands in at least powershell 5.1 and up

example:

```pwsh
$file = 'foo.txt'
git add $file
if ($LASTEXITCODE -ne 0) {
  throw "git command failed"
}
```

would the `-Environment` flag be able to solve this use case, as well? Or will these invocation examples need to be rewritten to use explicit calls to e.g. `Invoke-Command` or `Start-Process`?

**Comment by Velocity-** [comment]
It's absolutely bizarre that something so frequently needed gets to be an open issue for half a decade. Not that too much re: PowerShell makes sense, but this one just keeps bothering me.

Do you folks use your own shell? Surely it comes up daily?

**Comment by dkaszews** [comment]
@Velocity- With all due respect, I have used PowerShell as my daily driver shell for almost 5 years now, both on Windows and on Linux, and literally never run into this issue. Environment variables are not the only way to pass configuration - you can also use arguments, config files, etc. My workflows all use explicit arguments for everything (programming habit - avoid global state), so I rarely touch environment variables, and if so it's usually to set some defaults that can be overridden with arguments.

PowerShell is an open source product, if it bothers you "daily", then feel free to create a PR.

**Comment by dkaszews** [comment]
While I never had to override environment variables for a single command, I sometimes have to "push and pop" some other variables temporarily, usually when writing tests. So I would like the solution, probably a language construct, to also work for regular variables.

In C++ I wrote a little helper around `unique_ptr` with custom deleter to restore state at the end of the scope:

```cpp
auto x = 3;
{
    auto popGuard = tempChange(x, 5);  // need to give the finalizer object name to prevent immediate destruction
    ASSERT_EQ(x, 5);
}  // popGuard goes out of scope, its dtor restores original value
ASSERT_EQ(x, 3);
```

I think something similar could be written using C#'s `IDisposable` and `using tempChange(ref x, 5) { ... }` syntax. Does PowerShell support `using` blocks? If so, you just need to provide a helper.

**Comment by iSazonov** [comment]
@dkaszews Really API allows us to prepare environment block and use it to run subprocess. This doesn't change env block current process.
I made several PRs to solve this problem, but the code review process was so slow that I had to stop halfway through.
If anyone wants to continue this work, I'm willing to point the way and help.

**Comment by Velocity-** [comment]
@dkaszews 
>[..] and literally never run into this issue.
Good, I'm happy for you, but that tells me nothing because on the contrary, this issue has 30 comments and a whole lot of people showing interest in fixing it.

I'm lucky to not have to use PowerShell as my daily driver, or it would have 'daily driven' me insane. It's things like these that make it feel clumsy and it seems (based on what @iSazonov said) that attempts to improve it meet too much friction. Hence, the question "do you use your own shell".

So no, I'm not submitting a PR, because I simply don't care enough to fix someone else's product that I infrequently use. I'm simply bumping the tracker in hopes that the ergonomics of PS improve over time.

@iSazonov: Where did you get stuck? What's left? I couldn't find the branch in your (most recent) PS fork.

**Comment by dkaszews** [comment]
@Velocity- I did not mean to dismiss the importance of the feature for those who need it, just to answer your question - yes, people who contribute to PowerShell use it, but they may use it in different ways which mean they don't see some issues as critical. I personally like to focus on bugs and features that affect my workflows directly, and it looks like you see the same way with your "I simply don't care enough to fix someone else's product".

**Comment by iSazonov** [comment]
> Where did you get stuck? What's left? I couldn't find the branch in your (most recent) PS fork.

All I did was merged; no working branches exist.

**Comment by SteveL-MSFT** [comment]
Rather than introduce a syntax addition, I wonder if for most usage a smaller change would be sufficient:

```powershell
start-process ls -environment @{LS_COLORS='..."}
```

**Comment by iSazonov** [comment]
> Rather than introduce a syntax addition, I wonder if for most usage a smaller change would be sufficient:

That was what I was going for at the time, if you remember. :-) There is a detailed description in those PRs.

**Comment by Jackenmen** [comment]
This is a feature request, not a bug report, I don't think it can really become stale.

**Comment by rkeithhill** [comment]
Not just a feature request, but a reasonably high "upvoted" feature request.

**Comment by dkaszews** [comment]
Good fucking job, bot

**Comment by foxx1337** [comment]
> Rather than introduce a syntax addition, I wonder if for most usage a smaller change would be sufficient:
> 
> ```powershell
> start-process ls -environment @{LS_COLORS='..."}
> ```

I think it would be more useful for `Invoke-Expression`. So probably for all the cmdlets with the `Expression`, `Command` or `Process` nouns, if not others too.

**Comment by mklement0** [comment]
@foxx1337:

* [`Invoke-Expression` should generally be avoided](https://learn.microsoft.com/en-us/powershell/scripting/learn/deep-dives/avoid-using-invoke-expression). Aside from that, it simple accepts in _string_ form what you can execute _directly_, so this wouldn't help anyway. 

* `Start-Process` by now does have an `-Environment` parameter, which - while a welcome addition - is _not_ a solution to the problem at hand, as [previously explained](https://github.com/PowerShell/PowerShell/issues/3316#issuecomment-538063955). 

* _Conceivably_, `Invoke-Command` could be enhanced to support an `-Environment` parameter too, in a manner that _does_ provide stream integration, given that it is inherently _asynchronous_; however:

  * Unlike `Start-Process`, `Invoke-Command`'s focus isn't on calling _external programs_, i.e. not on creating _child processes_.

  * Something like `Invoke-Command -Environment @{LS_COLORS='...'} { foo ... }` is still a lot of ceremony compared to
 `LS_COLORS='...' foo ...`

**Comment by dkaszews** [comment]
Personally I don't mind a bit of ceremony. I'll take readability over obscure terseness any day of the week.

**Comment by jibbers42** [comment]
I'd prefer something quick and easy to bang out the command line and easy to remember without much fanfare. If I only had intentions of using the feature in scripts then I wouldn't much care how verbose it was.

**Comment by totkeks** [comment]
Was thinking about converting some bash scripts to PowerShell to improve their readability, but running a typical make command without that requires a lot of boilerplate.

What is the status and/or consensus on this one - will it come to PowerShell?

**Comment by mpawelski** [comment]
If there are no technical limitations or syntax conflicts I would really vote for making bash-like syntax just work in Powershell.

In my experience most of the shell examples in _any_ documentation are in (ba)sh, sometimes even on Microsoft docs. Luckily 80% of the time these examples works in Powershell just fine, 19% of other examples doesn't work because:
- command is multi-line and has `\` but pwsh requires `` ` ``
- command has [one-time env variable](https://www.gnu.org/software/bash/manual/bash.html#index-execution-environment:~:text=The%20environment%20for%20any%20simple%20command%20or%20function%20may%20be%20augmented%20temporarily%20by%20prefixing%20it%20with%20parameter%20assignments%2C%20as%20described%20in%20Shell%20Parameters.%20These%20assignment%20statements%20affect%20only%20the%20environment%20seen%20by%20that%20command.) (this issue) and pwsh not.
- other small syntax differences, like command has arguments that contains `@` `{` `}` characters
- other bash-specific features.

I would be great to decrease these chances of having to manually edit a command that you've copied, just because you use less popular shell.  It would be a huge usability win for new and long-time users of Powershell.

**Comment by MatejKafka** [comment]
@mpawelski
> If there are no technical limitations

Afaict, one of the blocking issues is that environment variables are global for the whole process, and PowerShell now supports thread jobs, meaning that there may be multiple commands running at once inside a single process – if you set an environment variable in multiple thread jobs, it will cause conflict, and re-architecting PowerShell to have separate env vars for each job would be quite hard.

However, I agree that even with this caveat, it would be worth it to add some form of support for this, since most users do not use thread jobs, and for native commands, the implementation may avoid the issue alltogether. Above, I proposed to use a fish-style `env` command to implement this without ambiguity (since `x=y` is a valid function name in PowerShell)):
```powershell
env TEST=1 cmd /c 'echo TEST=%TEST%'
```

This syntax is not compatible with Bash, but you only need to prefix the expression with `env` to make it work and PowerShell could have an error message to suggest the `env` command if you try to execute something that looks like Bash-style env var assignment.

**Comment by mklement0** [comment]
* Environment variables are primarily used for _inter-process_ communication, so that limiting the proposed POSIX-shell-like syntax to calls to _external (native) programs_ (child processes) makes sense to me.
  * In other words: an attempt to do something like `FOO=1 Get-ChildItem` should result in an error.
  * This also avoids the threads issue (assuming that the variables are truly only set in the environment of the _child_ process, which is the goal).

* On Unix-like platforms, the POSIX-mandated [`/usr/bin/env`](https://pubs.opengroup.org/onlinepubs/9799919799/utilities/env.html) utility already works the way your custom function does, e.g.:
      
       env TEST=1 sh -c 'echo $TEST'

* As for `x=y` being ambiguous, given that it is a valid function name in PowerShell:
  * I think a defensible way to resolve this is to tell users: use `&` to disambiguate in the unlikely event hat `x=y` is truly the name of a command.

---

In summary: 

* I think that implementing the POSIX-shell syntax in PowerShell too - for external programs only - is the way to go, and, within the constraints discussed above, I see no reasons that would make that infeasible.

* The suboptimal alternative is to provide a built-in `env` command on _Windows_ (to complement `/usr/bin/env` on Unix), as a _cmdlet_ with `env` as its alias.
  * This cmdlet should probably also implement the `-i` option of `/usr/bin/env`
  * Also, it should ideally only modify (pre-set) the _callee's_ environment, not also the _caller's_ (with the need for subsequent restoration, which is what your proof-of-concept does of technical necessity).

**Comment by dkaszews** [comment]
I think going with Unix syntax `x=y foo` is a bad idea:

1. Inconsistency with all variables set with `$x = y`
2. Inconsistency with environment variables set with `$env:x = y`
3. Inconsistency with no spaces required around `=`, which is a common pitfall with Unix beginners writing `x = y` and getting "x is not a command", plus the spaces may be added by existing formatters
4. No possibility to set non-literals, as `x=$y foo` would be interpreted as a string `"x=$y" foo` and not a command, while `& x=$y foo` would invoke command `"x=$y"` instead of `foo`

I would recommend instead implementing a `env` command for Windows, or, even better, do it the PowerShell way with `Invoke-Command -Environment @{ x = y } 'foo'` which solves all of the problems above.

**Comment by mklement0** [comment]
Number 4. is a non-issue, given that, e.g. `Write-Output FOO=$HOME` already works (argument-parsing mode); also, given that new syntax would be introduced, we'd have control over how to implement this.

Numbers 1. - 3. are fair points, but:

There are two major drivers for this feature:

* Convenience and concision

* Compatibility with POSIX shells, given the abundance of installation and sample command lines for the latter out there, as @mpawelski has pointed out.

Ensuring the latter is what makes the inconsistencies you mention unavoidable, just like we've already introduced _other_ inconsistencies for the sake of compatibilities with POSIX shells (automatic globbing (wildcard expansion) and tilde (`~`) expansion when calling external programs).

---

As for the `env` solution:

It occurred to me that implementing `env` (on Windows) as a _PowerShell_ command is _not_ an option, because `env` will have to behave _like any external (native) program_ with respect to its command-line parsing (not treating `,` as an array constructor, expanding `~`, assuming it [becomes a stable feature](https://learn.microsoft.com/en-us/powershell/scripting/learn/experimental-features?view=powershell-7.4#psnativewindowstildeexpansion), passing `--` through, ...)

Thus, there are only two options:

* Ship an actual `env.exe` binary on Windows.

* Bend the rules for an `env` command, which could then even be implemented as a _keyword_.

Neither option is great:

* Shipping `env.exe` has the advantage of not needing any changes in PowerShell itself, but it would be a first in terms of shipping an additional executable, and that executable would have to be discoverable via `$env:PATH`; this is currently _not_ a problem, but only due to the questionable practice of PowerShell prepending its `$PSHOME` value to `$env:PATH` on startup - which was done for historical reasons that no longer apply (see https://github.com/PowerShell/PowerShell/issues/24090#issuecomment-2267527201).

* Implementing `env` as a keyword would be awkward and preempt `/usr/bin/env` on Unix-like platforms.

**Comment by alexchandel** [comment]
7+ years and this is still impossible.  pwsh is not a shell replacement without this basic feature.  Why not even an Environment flag?

**Comment by mpawelski** [comment]
Just for reference, two other non-posix shell that use `key=value command` bash-inspired syntax:
- fish shell [since 3.1](https://fishshell.com/docs/current/faq.html#:~:text=Since%20fish%203.1%20you%20can%20set%20an%20environment%20variable%20for%20just%20one%20command%20using%20the%20key%3Dvalue%20some%20command%20syntax%2C%20like%20in%20other%20shells.)
- [NuShell](https://www.nushell.sh/book/environment.html#single-use-environment-variables:~:text=inspired%20by%20Bash%20and%20others)

Having such syntax  in Powershell, even just for native commands, would be great

**Comment by dkaszews** [comment]
I have been looking into implementing https://github.com/vim-test/vim-test/issues/832 and it would actually be nice to have some shell-agnostic syntax. For now it's `set var='value' ` in cmd, `var='value'` in POSIX shells and `$env:var = 'value'` in pwsh.

**Comment by aetos382** [comment]
In v7.4.0, `Start-Process -Environment @{ foo = ‘bar’ } app` became possible, but can this issue still not be closed?

By the way, I think the syntax `$env:foo=bar app` is problematic. This is because adding just one character—like `$env:foo=bar; app`—completely changes the meaning.

**Comment by seeker-3** [comment]
This is a big pain point for Node, having to install 3rd party ulilities like `cross-env` and `rimraf` just to get basic functionality working cross-platform. I would love to see something in PowerShell like the bash the native syntax `NODE_ENV=production node ...` or even an aliased `env` command `env NODE_ENV=production node ...`

**Comment by edgetools** [comment]
> By the way, I think the syntax $env:foo=bar app is problematic. This is because adding just one character—like $env:foo=bar; app—completely changes the meaning

so... identical to if you did

```
Foo=Bar; app
```

in sh/bash? lol
